python/test/unit/runtime/test_launch.py

```diff
@@ -33,3 +39,66 @@ def kernel(in_ptr0, out_ptr0, xnumel, XBLOCK: tl.constexpr):
         assert end - begin < 1000
     finally:
         tracemalloc.stop()
+
+
+def test_kernel_launch_latency() -> None:
+    def define_empty_kernel(file_path, num_tensor_args):
+        arg_str = ",".join([f"arg{i}: torch.Tensor" for i in range(num_tensor_args)])
+        arg_str += ", n_elements: int, BLOCK_SIZE: tl.constexpr"
+        func_str = f"""
+        import torch
+
+        import triton
+        import triton.language as tl
+
+        @triton.jit
+        def empty_kernel({arg_str}):
+            pass
+        """
+        with open(file_path, "w") as f:
+            f.write(textwrap.dedent(func_str))
+
+    def import_empty_kernel(file_path):
+        directory, filename = os.path.split(file_path)
+        module_name, _ = os.path.splitext(filename)
+        sys.path.insert(0, directory)
+
+        module = importlib.import_module(module_name)
+        empty_kernel = module.empty_kernel
+        return empty_kernel
+
+    def empty(*kernel_args: Tuple[torch.Tensor]):
+        first_arg = kernel_args[0]
+        n_elements = first_arg.numel()
+        grid = (triton.cdiv(n_elements, 1024),)
+        # Warmup
+        empty_kernel[grid](*kernel_args, n_elements, BLOCK_SIZE=1024, device=torch.cuda.current_device())
+        torch.cuda.synchronize()
```

Comment:
do I need this sync ?

Reply:
It's fine as long as you don't sync in the for loop